bot/exts/evergreen/trivia_quiz.py

```diff
@@ -202,27 +222,91 @@ def __init__(self, bot: Bot) -> None:
+        self.player_scores = defaultdict(int)  # A variable to store all player's scores for a bot session.
         self.game_player_scores = {}  # A variable to store temporary game player's scores.
 
         self.categories = {
             "general": "Test your general knowledge.",
             "retro": "Questions related to retro gaming.",
             "math": "General questions about mathematics ranging from grade 8 to grade 12.",
             "science": "Put your understanding of science to the test!",
+            "wikipedia": "Guess the title of random wikipedia passages.",
             "cs": "A large variety of computer science questions.",
             "python": "Trivia on our amazing language, Python!",
         }
 
+        self.get_wiki_questions.start()
+
+    def cog_unload(self) -> None:
+        """Cancel `get_wiki_questions` task when Cog will unload."""
+        self.get_wiki_questions.cancel()
+
+    @tasks.loop(hours=24.0)
+    async def get_wiki_questions(self) -> None:
+        """Get yesterday's most read articles from wikipedia and format them like trivia questions."""
+        error_fetches = 0
+        wiki_questions = []
+        # trivia_quiz.json follows a pattern, every new category starts with the next century.
+        start_id = 501
+        yesterday = datetime.strftime(datetime.now() - timedelta(1), '%Y/%m/%d')
+
+        while error_fetches < MAX_ERROR_FETCH_TRIES:
+            async with self.bot.http_session.get(url=WIKI_FEED_API_URL.format(date=yesterday)) as r:
+                if r.status != 200:
+                    error_fetches += 1
+                    continue
+                raw_json = await r.json()
+                articles_raw = raw_json["mostread"]["articles"]
+
+                for article in articles_raw:
+                    # Form the question, strategy:
+                    # - Check if `extract` is present in the article, if it doesn't then skip the article
+                    # Sometimes, the wikipedia title is given in the extract, therefore giving away the answer
+                    # therefore, we need to remove all cases of it
+                    question = article.get("extract")
+                    # multi letter words are formatted this way: Jason_Terror
+                    title = article["normalizedtitle"].replace("_", " ")
+
+                    if not question:
+                        continue
+                    elif not all(c.isalnum() for c in title):
+                        # If the title contains non alphabet/numerical(s) then
+                        # don't append it to the question and move on to the next.
+                        # Eg. "Jurassic World: Dominion (2022)" would be skipped.
+                        continue
+
+                    for word in re.split("[\s-]", title):  # noqa: W605
+                        word = word.strip(string.punctuation)
+                        secret_word = "\*"*len(word)  # noqa: W605
```

Comment:
You should avoid using `noqa`s if you can, as usually they make a good point (as they do here). Here you can use raw strings, e.g. `r"[\s-]"`  to make it clear `\s` isn't an escape sequence meant to be interpreted by python